File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.js
function divisorCount(num) {
  let count = 0;
  for (let i = 1; i <= num; i++) {
    if (num % i === 0) count++;
  }
  return count;
}

function solution(left, right) {
  let answer = 0;
  for (let i = left; i <= right; i++) {
    if (divisorCount(i) % 2 === 0) {
      // 짝수
      answer += i;
    } else {
      answer -= i;
    }
  }

  return answer;
}

console.log(solution(13, 17));

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def divisor_count(num: int) -> int:
    count = 0
    root = int(num ** 0.5)
    for i in range(1, root + 1):
        if num % i == 0:
            count += 1 if i == num // i else 2
    return count

def solution(left: int, right: int) -> int:
    total = 0
    for value in range(left, right + 1):
        if divisor_count(value) % 2 == 0:
            total += value
        else:
            total -= value
    return total
